File: Remove Duplicates from Sorted Array II/Remove Duplicates from Sorted Array II.py
class Solution:
    # @param A a list of integers
    # @return an integer
    def removeDuplicates(self, A):
        newl=0
        for i in range(len(A)):
            if newl==0 or i==len(A)-1 or A[i]!=A[i-1] or A[i]==A[i-1] and A[i]!=A[i+1]:
                A[newl]=A[i]
                newl+=1
        return newl

File: Remove Duplicates from Sorted Array II/Remove Duplicates from Sorted Array II.java
public class Solution {
    public int removeDuplicates(int[] A) {
        int newl = 0;
        for(int i = 0; i < A.length; i++) {
            if(i==A.length-1||newl == 0|| A[i] != A[i-1]||(A[i]==A[i-1]&&A[i]!=A[i+1])){
               A[newl++] = A[i]; 
            }
        }
        return newl;
    }
}
